### PR TITLE
Ch 1345: Remove "personalize existing"

### PR DESCRIPTION
**Pull request by eshtadc**
Also replace the "add variation set" with a modal to display personalize blocks and personalize elements options.

### CODE REVIEW

File: acquia_lift.admin.inc [hunk 1]
```diff
@@ -1233,6 +1233,55 @@ function acquia_lift_page_variation_details_form_ajax_callback($form, &$form_sta
...
+  ctools_include('modal');
+  ctools_modal_add_js();
+
+  $path = drupal_get_path('module', 'acquia_lift');
+  $output = theme('acquia_lift_type_list', array(
+    'id' => 'acquia-lift-option-set-type-list',
+    'items' => array(
+      'personalize-block' => array(
+        'title' => t('Personalized block'),
+        'description' => t('Quisque malesuada placerat nisl. Phasellus volutpat, metus eget egestas mollis, lacus lacus blandit dui, id egestas quam mauris ut lacus.'),
```
Comment (katbailey) [inline acquia_lift.admin.inc:1259]: I guess this is part of " there are some design texts to be changed" :)

File: acquia_lift.admin.inc [hunk 2]
```diff
@@ -1233,6 +1233,55 @@ function acquia_lift_page_variation_details_form_ajax_callback($form, &$form_sta
...
+        'path' => 'admin/structure/personalize/variations/personalize-blocks/add',
+        'modal' => FALSE,
+        'logo' => theme('image', array(
+          'path' => $path . '/images/goal-type-predefined.png',
+          'alt' => t('Pre-existing goal'),
+          'title' => t('Select this option to create a goal from a pre-existing visitor action.'),
```
Comment (katbailey) [inline acquia_lift.admin.inc:1265]: And this - pointing it out just in case...

Comment (eshtadc) [inline acquia_lift.admin.inc:1265]: Yup - this whole list will be replaced with the correct title, description, and image details once we have some idea what those will be..... Seems to me that at last discussion time, the base terms were still up for a lot of debate.

File: acquia_lift.module [hunk 3]
```diff
@@ -2238,6 +2225,49 @@ function acquia_lift_personalize_elements_confirmation_form_submit($form, &$form
...
+/**
+ * Submit handler for altered personalize_elements_form.
+ */
+function acquia_lift_personalize_blocks_form_submit($form, &$form_state) {
+  $redirect = $form_state['redirect'];
+  if ($redirect === FALSE) {
+    return;
+  }
+  // A destination string overrides redirects set in the form state.
+  $params = drupal_get_query_parameters();
```
Comment (katbailey) [inline acquia_lift.module:2243]: Might make sense to use drupal_get_destination here... https://api.drupal.org/api/drupal/includes%21common.inc/function/drupal_get_destination/7

File: acquia_lift.ui.inc [hunk 4]
```diff
@@ -74,9 +74,20 @@ function acquia_lift_build_page(&$page) {
...
     $pending_agents = acquia_lift_get_pending_agents();
     if (!empty($pending_agents)) {
-      $settings['acquia_lift']['pendingMessage'] = t('The following campaigns are being synced to Lift and will be started shortly: %agents', array('%agents' => implode(', ', $pending_agents)));
+      $settings['acquia_lift']['pendingMessage'][] = t('The following campaigns are being synced to Lift and will be started shortly: %agents', array('%agents' => implode(', ', $pending_agents)));
       $page['page_top']['#attached']['library'][] = array('acquia_lift', 'acquia_lift.messagebox');
     }
+    // Add any pending messages from the query string.
+    $params = drupal_get_query_parameters();
+    if (!empty($params['pm'])) {
+      list($type, $details) = explode('|', $params['pm']);
```
Comment (katbailey) [inline acquia_lift.ui.inc:83]: With a query param like 'pm' there's the possibility of it being set by some other code. We should probably at least check that it contains the '|' character before trying to parse it.

File: acquia_lift.module [hunk 5]
```diff
@@ -2238,6 +2225,49 @@ function acquia_lift_personalize_elements_confirmation_form_submit($form, &$form
...
+  }
+  // Convert the existing redirect into a redirect that accepts query parameters.
+  if (is_string($redirect)) {
+    $redirect = array(
+      $redirect,
+      array('query' => array())
+    );
+  }
+  // Add a message to be shown.
+  $redirect[1]['query']['pm'] = 'new_block|' . $form_state['values']['title'];
```
Comment (katbailey) [inline acquia_lift.module:2266]: Neat! :)

File: acquia_lift.module (hunk 3 above)
Comment (eshtadc) [inline acquia_lift.module:2243]: Thank you - I had no idea this function existed :-)

**Pull request closed without merging**